src/protocol/sort/generate_permutation.rs

```diff
@@ -117,7 +119,7 @@ pub(super) async fn malicious_shuffle_and_reveal_permutation<F: Field>(
     Ok(RevealedAndRandomPermutations {
         revealed: revealed_permutation,
         randoms_for_shuffle: random_permutations_for_shuffle,
-    })
+    })*/
```

Comment:
I meant to leave a comment here when I opened the PR. The trouble with this code is that `MaliciousValidator::validate` takes the validator by value, which is incompatible with `m_ctx` having an active borrow.

One idea that I had for dealing with this is to have `DowngradeMalicious::downgrade` take a context as an argument. However, I'm not sure I like that as a solution, since many kinds of values can be downgraded without a context.

So I'm still thinking about how best to handle this, and I'm open to suggestions.